### PR TITLE
Refactoring code

### PR DESCRIPTION
**Pull request by sattellite**
Code must be refactored for better understanding and clear code:

- [x] Rename `Ctx` to `Req` for correct naming
- [x] Move code to internal and cmd directories for [better structure](https://github.com/golang-standards/project-layout/tree/master/cmd)
- [x] Explore for usage of `context.Context` (best introducing [video](https://www.youtube.com/watch?v=LSzR0VEraWw))
- [ ] ~~Separate package for handling commands messages from telegram~~
- [x] Wrap all errors in packages

### CODE REVIEW

File: internal/bot/bot.go [hunk 1]
```diff
@@ -57,58 +45,55 @@ func main() {
...
-		Bot:    bot,
+		API:    bot,
+		Log:    log,
 	}
+}
 
-	log.Infof("Authorized on telegram account @%s", bot.Self.UserName)
-
+// Start starts polling for messages for bot
+func Start(bot *t.Bot) {
```
Comment (sah4ez) [inline internal/bot/bot.go:64]: Maybe implement method `Start` in the `*t.Bot` like:
`func (b *Bot) Start() (err error) { return }`
And using in the main looks like 
```
//...
bot := NewBot(config)
err = bot.Start()
//...
```

File: internal/bot/bot.go [hunk 2]
```diff
@@ -57,58 +45,55 @@ func main() {
...
 	u := tg.NewUpdate(0)
 	u.Timeout = 60
 
-	updates, err := bot.GetUpdatesChan(u)
+	updates, err := bot.API.GetUpdatesChan(u)
+
+	if err != nil {
+		bot.Log.Errorf("Failed get updates from Telegram. %v", err)
+		os.Exit(4)
```
Comment (sah4ez) [inline internal/bot/bot.go:72]: Why status code is `4`?
https://tldp.org/LDP/abs/html/exitcodes.html

File: internal/bot/bot.go [hunk 3]
```diff
@@ -57,58 +45,55 @@ func main() {
...
 		switch {
 		case update.Message.IsCommand():
-			log.WithFields(logrus.Fields{
-				"requestID": ctx.RequestID,
+			bot.Log.WithFields(logrus.Fields{
+				"requestID": req.ID,
 				"user":      update.Message.From,
 			}).Infof("Command request %s %s", update.Message.Command(), update.Message.CommandArguments())
-			go command(ctx, update.Message)
+			go command.Handle(req, update.Message)
```
Comment (sah4ez) [inline internal/bot/bot.go:90]: Maybe move implementation `Handle` to the `t *t.Bot`?

File: internal/bot/bot.go (hunk 2 above)
Comment (sattellite) [reply internal/bot/bot.go]: I just used incremental value for exit code values. No particular reason.
Fixed in 9fd8bdf67a

**Comment by sattellite** [comment]
> I think, need to do some fixes.

Implemented fixes:
- all functions is methods of `bot` internal package
- `context.Context` integration

Check please.

File: internal/bot/bot.go [hunk 4]
```diff
@@ -51,64 +55,67 @@ func main() {
...
 		switch {
 		case update.Message.IsCommand():
-			log.WithFields(logrus.Fields{
-				"requestID": ctx.RequestID,
+			b.Log.WithFields(logrus.Fields{
+				"requestID": req.ID,
 				"user":      update.Message.From,
 			}).Infof("Command request %s %s", update.Message.Command(), update.Message.CommandArguments())
-			go command(ctx, update.Message)
+			go b.HandleCommand(ctx, update.Message)
```
Comment (sah4ez) [inline internal/bot/bot.go:107]: looks like you can didn't use ctx here, you can just put request as argument. And use ctx for deadline/cancelation request.

File: internal/bot/bot.go [hunk 5]
```diff
@@ -51,64 +55,67 @@ func main() {
...
 		// Create context for request
 		reqTime := time.Now()
-		ctx := Ctx{
-			App:         app,
-			Log:         log,
-			RequestID:   reqTime.UnixNano() / 1000,
-			RequestTime: reqTime,
+		req := BotRequest{
+			ID:   reqTime.UnixNano() / 1000,
+			Time: reqTime,
```
Comment (sah4ez) [inline internal/bot/bot.go:96]: looks as dead code)

File: internal/bot/bot.go (hunk 4 above)
Comment (sattellite) [reply internal/bot/bot.go:107]: Telegram Bot API don't cancel requests. I not need in deadline and cancelation here.

**Pull request merged**